tests: flash: nxp: mimxrtxxx series: all flash related application meet build failure

**Describe the bug**

<html>
<body>

CASE_NAME | ZEPHYR_VERSION | RESULT | PLATFORM | TEST_PATH
-- | -- | -- | -- | --
drivers.flash.common.default.flash_driver.flash_page_layout | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | drivers.flash
drivers.flash.common.default.flash_driver.flash_flatten | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | drivers.flash
drivers.flash.common.default.flash_driver.flash_fill | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | drivers.flash
drivers.flash.common.default.flash_driver.flash_erase | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | drivers.flash
drivers.flash.common.default.flash_driver.flash_copy | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | drivers.flash
drivers.flash.common.default.flash_driver.read_unaligned_address | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | drivers.flash
drivers.flash.common.default.flash_driver.get_size | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | drivers.flash
bootloader.mcuboot | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | bootloader.mcuboot
mgmt.mcumgr.fs.mgmt.hash.supported.crc32.fs_mgmt_hash_supported.supported | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | mgmt.mcumgr
mgmt.mcumgr.fs.mgmt.hash.supported.sha256.fs_mgmt_hash_supported.supported | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | mgmt.mcumgr
mgmt.mcumgr.fs.mgmt.hash.supported.all.fs_mgmt_hash_supported.supported | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | mgmt.mcumgr
secure_storage.psa.crypto.secure_storage.secure_storage_psa_crypto.persistent_key_usage | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | secure_storage.psa
secure_storage.psa.crypto.secure_storage.secure_storage_psa_crypto.its_caller_isolation | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | secure_storage.psa
secure_storage.psa.its.secure_storage.secure_storage_psa_its.all_offsets | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | secure_storage.psa
secure_storage.psa.its.secure_storage.secure_storage_psa_its.all_sizes | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | secure_storage.psa
secure_storage.psa.its.secure_storage.secure_storage_psa_its.max_num_entries | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | secure_storage.psa
secure_storage.psa.its.secure_storage.secure_storage_psa_its.write_once_flag | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | secure_storage.psa
secure_storage.psa.its.secure_storage.custom.transform.secure_storage_psa_its.all_offsets | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | secure_storage.psa
secure_storage.psa.its.secure_storage.custom.transform.secure_storage_psa_its.all_sizes | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | secure_storage.psa
secure_storage.psa.its.secure_storage.custom.transform.secure_storage_psa_its.max_num_entries | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | secure_storage.psa
secure_storage.psa.its.secure_storage.custom.transform.secure_storage_psa_its.write_once_flag | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | secure_storage.psa
secure_storage.psa.its.secure_storage.custom.store.secure_storage_psa_its.all_offsets | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | secure_storage.psa
secure_storage.psa.its.secure_storage.custom.store.secure_storage_psa_its.all_sizes | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | secure_storage.psa
secure_storage.psa.its.secure_storage.custom.store.secure_storage_psa_its.max_num_entries | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | secure_storage.psa
secure_storage.psa.its.secure_storage.custom.store.secure_storage_psa_its.write_once_flag | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | secure_storage.psa
secure_storage.psa.its.secure_storage.custom.both.secure_storage_psa_its.all_offsets | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | secure_storage.psa
secure_storage.psa.its.secure_storage.custom.both.secure_storage_psa_its.all_sizes | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | secure_storage.psa
secure_storage.psa.its.secure_storage.custom.both.secure_storage_psa_its.max_num_entries | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | secure_storage.psa
secure_storage.psa.its.secure_storage.custom.both.secure_storage_psa_its.write_once_flag | v4.0.0-1231-g6c7b38f5fa1c | Build failure | mimxrt685_evk/mimxrt685s/cm33 | secure_storage.psa

</body>
</html>

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - mimxrt685_evk/mimxrt685s/cm33 and mimxrt595_evk/mimxrt595s/cm33
 - What have you tried to diagnose or workaround this issue?
 - git bisect
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 
```
6efba38e970d401582159614ed35dbfa93a89730 is the first bad commit
commit 6efba38e970d401582159614ed35dbfa93a89730
Author: Dominik Ermel <dominik.ermel@nordicsemi.no>
Date:   Wed Jul 14 10:44:53 2021 +0000

    drivers/flash/flexspi_x25um51345g: Add flash_get_size API call
    
    The commit adds implementation of flash_get_size.
    
    Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

 drivers/flash/flash_mcux_flexspi_mx25um51345g.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)
```


**To Reproduce**

```
scripts/twister -p mimxrt595_evk/mimxrt595s/cm33 -T tests/drivers/flash/common
```

**Expected behavior**

build pass

**Impact**

flash features


**Logs and console output**

```
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/flash/flash_mcux_flexspi_mx25um51345g.c: In function 'flash_flexspi_nor_get_size':
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/flash/flash_mcux_flexspi_mx25um51345g.c:505:23: error: invalid use of undefined type 'const struct flash_flexspi_nor_config'
  505 |         *size = config->size;
      |                       ^~
In file included from 
```

**Environment (please complete the following information):**

 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: v4.0.0-1231-g6c7b38f5fa1c

--- a/drivers/flash/flash_mcux_flexspi_mx25um51345g.c
+++ b/drivers/flash/flash_mcux_flexspi_mx25um51345g.c
@@ -67,7 +67,7 @@ struct flash_flexspi_nor_data {
 	const struct device *controller;
 	flexspi_device_config_t config;
 	flexspi_port_t port;
-	uint64_t *size;
+	uint64_t size;
 	struct flash_pages_layout layout;
 	struct flash_parameters flash_parameters;
 };
@@ -500,9 +500,9 @@ static const struct flash_parameters *flash_flexspi_nor_get_parameters(
 
 static int flash_flexspi_nor_get_size(const struct device *dev, uint64_t *size)
 {
-	const struct flash_flexspi_nor_config *config = dev->config;
+	const struct flash_flexspi_nor_data *data = dev->data;
 
-	*size = config->size;
+	*size = data->size;
 
 	return 0;
 }

